sympy/matrices/expressions/tests/test_matrix_exprs.py

```diff
@@ -213,3 +213,32 @@ def test_MatrixElement_doit():
     u = MatrixSymbol('u', 2, 1)
     v = ImmutableMatrix([3, 5])
     assert u[0, 0].subs(u, v).doit() == v[0, 0]
+
+
+def test_identity_powers():
+    M = Identity(n)
+    assert MatPow(M, 3).doit() == M**3
```

Comment:
you should replace `M**3` -> `M`, since `MatPow(M, 3).doit() == M**3` was already `True`. So i guess this currently does not really check what change has been made.
Similarly for other tests replace `M**n` -> `M`(wherever valid), for the same reason.